### DISCUSSION TITLE
Setting to use blur filter for yellow areas in vegetation?

### DISCUSSION THREAD

**Discussion by stefankinell** [karttapullautin/karttapullautin#55] [Q&A] [answered]
In the .ini-file I have learned to play around with

```
# green raster image filtering with median filter. Two rounds
# use 1 to do no filtering.
medianboxsize=9
medianboxsize2=1

```
In the last rerun of vegetation for Sweden I used 
```
medianboxsize=15
medianboxsize2=9

```
and with that setting I manage to get green areas that are more consistent and not so dotted as the original. It is less acurate if you think of how it is in each square meter, but I like the looks of it better and it is more similar to how orienteering maps are drawn.

But, I can not figure out if there is a similar setting that would influence the yellow boxes. I can only find two yellow parameters in the ini-file:
```
## yellow parameters
### hits below this will be calculated as yellow
yellowheight=0.9  

### how big part or the points must be below yellowheight to trigger yellow
yellowthresold=0.7 
```
and I can not understand if any of the green parameters also influences the yellow areas.

Here is an extract from a demo area of the new vegetation layer
![image](https://github.com/rphlo/rusty-pullauta/assets/59049892/afa11078-df90-4306-a9a1-520732307b81)
It is taken from https://kartor.gokartor.se/dev/#15/56.1442/15.8316 (an url that works now but perhaps will not work a few months from now...)

As can see in the image the green seems to have a more detailed resolution and the yellow has more larger pixels. I understand that the green is somewhat smoothed out. But I think it is much nicer - and by that makes the maps more readable.

**Comment by rphlo** [comment]
The vegetation (greens) is indeed smoothed out with a median blur filter and there is no equivalent for the yellows as I recall.
It could be implemented quite easily, with default being no bluring to maintain compatibility with old version.

**Comment by stefankinell** [comment]
Thanks, then I was right in my assumptions. I know you know the script quite well so I figured it was easier to ask.

If it is easy to implement for yellow I think it would be a nice feature to implement.

**Comment by rphlo** [comment]
For reference,
Yellows file saving occurs here:
https://github.com/rphlo/rusty-pullauta/blob/6dd510e2edb2272b5df795ea67d696e0a5210692/src/main.rs#L4262

And greens blurring occurs here:
https://github.com/rphlo/rusty-pullauta/blob/6dd510e2edb2272b5df795ea67d696e0a5210692/src/main.rs#L4324

**Comment by rphlo** [comment, marked as answer]
See #56

**Reply by stefankinell** [reply]
Nice!
I'll try it out next week.

**Reply by rphlo** [reply]
@stefankinell Have you had the chance to test?

**Reply by stefankinell** [reply]
I have tested it in terms of seeing it works technically. I ran a few larger areas and it did what I expected.

What I would like to do is to run some more tests and tweeking my settings. Right now I get nicely shaped yellow areas when I set 
```
medianboxsize=16
medianboxsize2=9

```
But it also means it is removing a lot of smaller yellow areas. I liked that approach for the green colours, but have not decided yet if I like it as much for the yellow. So I will play around with the settings and see where I get a good middle ground. Perhaps having different settings for yellow and green box sizes as an alternative.
